crates/flox/src/commands/environment.rs

```diff
@@ -64,11 +64,18 @@ impl Edit {
             // If not provided with new manifest contents, let the user edit the file directly
             // via $EDITOR (as long as `flox edit` was invoked interactively).
             None => {
-                let editor = std::env::var("EDITOR").context("$EDITOR not set")?;
-                if !Dialog::can_prompt() {
-                    bail!("Can't open editor in non interactive context");
-                }
-                let path = environment.manifest_path();
+                let editor = std::env::var("EDITOR")
+                    .or(std::env::var("VISUAL"))
+                    .context("no editor found; neither EDITOR nor VISUAL are set")?;
+                // TODO: check for interactivity before allowing the editor to be opened
+                let manifest_path = environment.manifest_path();
+                // Make a copy of the manifest for the user to edit so failed edits aren't left in
+                // the original manifest. You can't put creation/cleanup inside the `edited_manifest_contents`
+                // method because the temporary manifest needs to stick around in case the user wants
+                // or needs to make successive edits without starting over each time.
+                let tmp_manifest_path = flox.temp_dir.join("tmp_manifest.nix");
```

Comment:
(can be ignored/just a thought) if we expose the transaction model, we could modify the temporary manifest that's already been created for us, rather than having to re-implement a mini-transaction